src/types/params.ts

```diff
@@ -43,11 +43,11 @@ export type ExtractPathParamType<
   TParams extends Record<string, Param | undefined>
 > = TParam extends `?${infer OptionalParam}`
   ? OptionalParam extends keyof TParams
-    ? ExtractParamType<TParams[OptionalParam]> | undefined
-    : string | undefined
+    ? TParams[OptionalParam] | undefined
+    : StringConstructor | undefined
```

Comment:
note Path and Query params are expected to extend `Param` not the return value of the param